website/docs/r/upgrade_prepare.html.markdown

```diff
@@ -0,0 +1,58 @@
+---
+subcategory: "Beta"
+layout: "nsxt"
+page_title: "NSXT: nsxt_upgrade_prepare"
+description: A resource to prepare for the upgrade of NSXT cluster.
+---
+
+# nsxt_upgrade_prepare
+
+This resource provides a method for preparing the upgrade of NSXT cluster,
+it will upload upgrade bundle to NSX and run upgrade prechecks.
+If there are errors in precheck result, the creation of resource will fail.
+Precheck warnings will be listed in the `failed_prechecks` field of terraform state,
+user can use the nsxt_upgrade_precheck_acknowledge resource to acknowledge these
+warnings.
+When using this resource to prepare for NSXT upgrade, the username and password
+for NSXT must be provided in the nsxt provider config.
+
+## Example Usage
+
+```hcl
+resource "nsxt_upgrade_prepare" "test" {
+  upgrade_bundle_url    = "http://url-to-upgrade-bundle.mub"
+  precheck_bundle_url   = "http://url-to-precheck-bundle.pub"
+  accept_user_agreement = true
+  bundle_upload_timeout = 1600
+  uc_upgrade_timeout    = 600
+  precheck_timeout      = 1600
+  version               = "4.1.2"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `upgrade_bundle_url` - (Required) The url to download the Manager Upgrade bundle.
+* `version` - (Optional) Target upgrade version for NSX, format is x.x.x..., should include at least 3 digits, example: 4.1.2
+* `precheck_bundle_url` - (Optional) The url to download the Precheck bundle. This field is required and only applied to NSX version >= 4.1.1
```

Comment:
nit: This field is required if NSX >= 4.1.1

The current statement creates a bit of confusion on whether the parameter is required or optional

Reply:
done